lusidtools/cocoon/utilities.py

```diff
@@ -260,27 +260,40 @@ def set_attributes_recursive(
     # For each of the attributes to populate
     for key in list(populate_attributes):
 
+        # Get the attribute type
+        attribute_type = obj_attr[key]
+
         # If it is an additional attribute, populate it with the provided values and move to the next attribute
         if key in list(additional_attributes.keys()):
-            obj_init_values[key] = additional_attributes[key]
-            continue
+            # Handle identifiers provided within instrument definition (e.g. 'Bond', 'Future', etc.)
+            if (key, attribute_type) == ("identifiers", "dict(str, str)"):
```

Comment:
Ahhh very nice, great pickup that we are considering `identifiers` to be an exception to how we process most attributes,  clearly with these `LusidInstruments` that exception is broken. This seems to be a sensible way of handling the exception to the exception.